src/fidesops/graph/data_type.py

```diff
@@ -131,3 +158,44 @@ class DataType(Enum):
     float = FloatTypeConverter()
     boolean = BooleanTypeConverter()
     object_id = ObjectIdTypeConverter()
+    object = ObjectTypeConverter()
+    no_op = NoOpTypeConverter()
+
+
+_data_type_names: Set[str] = {
+    "string",
+    "integer",
+    "float",
+    "boolean",
+    "object_id",
+    "object",
+}
```

Comment:
I like how you specify arrays in the yaml, but I don't understand why arrays have to be treated differently internally.  To me, it is confusing to not have array be an allowed `_data_type_name` and then we have a sub data type name that is "string" or similar

Reply:
Initially there was a separate Array type, to go along with Scalar and Object types. What you then run into very quickly is a kind of structural mess - 
Think of the name property,e.g
```
  scalar:
      name: "id", type:"int"
  object:
     name:"parent", fields: [name:"child*", type: "int"]  * this is referred to as "parent.child"
  array?
    name:"id": field; ["name": ? ] 
   ```
     Here if the array will share properties with the wrapped field it might as well just be the wrapped field. Additionally you can have an array of scalars, and an array of objects, so now you need 4 types, which seems overdone, especially as arrays are entirely identity to what they would wrap other than the distinction that they represent an array.

Reply:
I see what you're saying -  I wonder though if we'll run into instances where datasources allow arrays to contain different types.

Reply:
That seems pretty fringe; I think this falls under the heading of slightly strange database usage, and we've run into the question a few times as to what level of "unusual" database use to support.

I'm pretty sure, for example, that you can store something like this: 
`{1,"A",{"x":2}}`
In both mongo and dynamo.
I suppose it's possible that someone might have a datastore where these were typed by position - i.e. "the second element is always a string" so you'd kind of have a logical map by int value, but do we really want to support that? 

More likely, I think, that if you had an array like that, where it's just a grab-bag of stuff, there's not ordering. I think for type information we'd want some kind of type-dependent logic like "mask whatever the item # that was a string with string masking behavior", which sounds a bit like the union types Sean was suggesting. 

Maybe another way of attacking this problem, e.g. in masking a heterogeneous array, would be to pull the whole array, apply masking to each element by appropriate type.

Reply:
makes sense, good masking suggestions -